### PR TITLE
Impliment auto update

### PR DESCRIPTION
**Pull request by jehy**
Hi! I implemented auto update, using code from PokeScanner. It looks like this:

screenshot 19 2016 12_28_41

It should be fine but please double check result :)

### CODE REVIEW

File: app/src/main/java/com/kamron/pogoiv/updater/AppUpdateDialog.java [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+                    install.setFlags(Intent.FLAG_ACTIVITY_NEW_TASK);
+                    install.setDataAndType(uri, manager.getMimeTypeForDownloadedFile(downloadId));
+                    ctxt.startActivity(install);
+                    ctxt.unregisterReceiver(this);
+                }
+            };
+            //register receiver for when .apk download is complete
+            context.registerReceiver(onComplete, new IntentFilter(DownloadManager.ACTION_DOWNLOAD_COMPLETE));
+        } catch (Exception e) {
+            System.out.println("We have an error houston");
```
Comment (agrajaghh) [inline app/src/main/java/com/kamron/pogoiv/updater/AppUpdateDialog.java]: log.e ?

**Comment by jehy** [comment]
Yup, it can be done if request is merged. For now I just wanted to make a quick implementation.
Also an option to disable auto update can be added - but it's for later.

**Comment by nahojjjen** [comment]
I think it would be good to test this feature on github with a dummy update before we release the auto update on reddit.

**Comment by nahojjjen** [comment]
As for exception handling, I don't know much about crashlytics, but it's used in other catch blocks, and the crash reports are useful. Here's how its used in another place:


            Crashlytics.log("Exception thrown while getting version name");

            Crashlytics.logException(e);

            Log.e(TAG, "Error while getting version name", e);

**Comment by rhari991** [comment]
You can test this out locally as well, just change the versionCode in your build.gradle file to something less than the highest version on GitHub (2.1.0 right now), and try running it.

**Comment by sarav** [comment]
Hey, due to some mistake of mine the master branch had to be rewound in history. You might have to rebase or cherry-pick your changes and push you branch again and make sure it only contains your changes and not old crap from me.

**Comment by farkam135** [comment]
Ready to merge, please fix any conflicts!

**Comment by jehy** [comment]
Ok, I rebased and fixed conflicts, ready to merge.

**Pull request merged**